pkg/jobs/job_info_utils.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package jobs
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/sql/isql"
+	"github.com/cockroachdb/errors"
+	"github.com/klauspost/compress/gzip"
+)
+
+const bundleChunkSize = 1 << 20 // 1 MiB
+const finalChunkSuffix = "#_final"
+
+// WriteChunkedFileToJobInfo will break up data into chunks of a fixed size, and
+// gzip compress them before writing them to the job_info table
+func WriteChunkedFileToJobInfo(
+	ctx context.Context, filename string, data []byte, txn isql.Txn, jobID jobspb.JobID,
+) error {
+	jobInfo := InfoStorageForJob(txn, jobID)
+
+	var chunkCounter int
+	var chunkName string
+	for len(data) > 0 {
+		chunkSize := bundleChunkSize
+		chunk := data
+		if len(chunk) > chunkSize {
+			chunkName = fmt.Sprintf("%s#%04d", filename, chunkCounter)
```

Comment:
so we cant have more than 999 chunks, given the chunk name. Should this function through an error if we reach that state?

Reply:
good point, but I think its 9999 chunks, which is ~10gigs with each chunk being 1MB. Added an assertion failure for if we ever hit this limit.